llvm/lib/Target/RISCV/RISCVInstrInfoQ.td

```diff
@@ -0,0 +1,168 @@
+    defm FMAX_Q : FPALU_rr_m<0b0010111, 0b001, "fmax.q", Ext, Commutable = 1>;
+  }
+
+  defm FCVT_S_Q : FPUnaryOp_r_frm_m<0b0100000, 0b00011, Ext, Ext.F32Ty,
+                                    Ext.PrimaryTy, "fcvt.s.q">,
+                  Sched<[WriteFCvtF128ToF32, ReadFCvtF128ToF32]>;
+
+  defm FCVT_Q_S : FPUnaryOp_r_frmlegacy_m<0b0100011, 0b00000, Ext,
+                                          Ext.PrimaryTy, Ext.F32Ty, "fcvt.q.s">,
+                  Sched<[WriteFCvtF32ToF128, ReadFCvtF32ToF128]>;
+
+  defm FCVT_D_Q : FPUnaryOp_r_frm_m<0b0100001, 0b00011, Ext, Ext.F64Ty,
+                                    Ext.PrimaryTy, "fcvt.d.q">,
+                  Sched<[WriteFCvtF128ToF64, ReadFCvtF128ToF64]>;
+
+  defm FCVT_Q_D : FPUnaryOp_r_frmlegacy_m<0b0100011, 0b00001, Ext,
+                                          Ext.PrimaryTy, Ext.F64Ty, "fcvt.q.d">,
+                  Sched<[WriteFCvtF64ToF128, ReadFCvtF64ToF128]>;
+
+  let SchedRW = [WriteFCmp128, ReadFCmp128, ReadFCmp128] in {
+    defm FEQ_Q : FPCmp_rr_m<0b1010011, 0b010, "feq.q", Ext, Commutable = 1>;
+    defm FLT_Q : FPCmp_rr_m<0b1010011, 0b001, "flt.q", Ext>;
+    defm FLE_Q : FPCmp_rr_m<0b1010011, 0b000, "fle.q", Ext>;
+  }
+
+  let mayRaiseFPException = 0 in 
+  defm FCLASS_Q : FPUnaryOp_r_m<0b1110011, 0b00000, 0b001, Ext, GPR,
+                                Ext.PrimaryTy, "fclass.q">,
+      Sched<[WriteFClass128, ReadFClass128]>;
+
+  let IsSignExtendingOpW = 1 in 
+  defm FCVT_W_Q : FPUnaryOp_r_frm_m<0b1100011, 0b00000, Ext, GPR,
+                                    Ext.PrimaryTy, "fcvt.w.q">,
+      Sched<[WriteFCvtF128ToI32, ReadFCvtF128ToI32]>;
+
+  let IsSignExtendingOpW = 1 in 
+  defm FCVT_WU_Q : FPUnaryOp_r_frm_m<0b1100011, 0b00001, Ext, GPR,
+                                     Ext.PrimaryTy, "fcvt.wu.q">,
+      Sched<[WriteFCvtF128ToI32, ReadFCvtF128ToI32]>;
+
+  let mayRaiseFPException = 0 in 
+  defm FCVT_Q_W : FPUnaryOp_r_frm_m<0b1101011, 0b00000, Ext,
+                                    Ext.PrimaryTy, GPR, "fcvt.q.w">,
+      Sched<[WriteFCvtI32ToF128, ReadFCvtI32ToF128]>;
+
+  let mayRaiseFPException = 0 in 
+  defm FCVT_Q_WU : FPUnaryOp_r_frm_m<0b1101011, 0b00001, Ext,
+                                     Ext.PrimaryTy, GPR, "fcvt.q.wu">,
+      Sched<[WriteFCvtI32ToF128, ReadFCvtI32ToF128]>;
+
+} // foreach Ext = QExts
+
+foreach Ext = QExtsRV64 in {
+  defm FCVT_L_Q : FPUnaryOp_r_frm_m<0b1100011, 0b00010, Ext, GPR, Ext.PrimaryTy,
+                                    "fcvt.l.q", [IsRV64]>,
+                  Sched<[WriteFCvtF128ToI64, ReadFCvtF128ToI64]>;
+
+  defm FCVT_LU_Q : FPUnaryOp_r_frm_m<0b1100011, 0b00011, Ext, GPR,
+                                     Ext.PrimaryTy, "fcvt.lu.q", [IsRV64]>,
+                   Sched<[WriteFCvtF128ToI64, ReadFCvtF128ToI64]>;
+
+  let mayRaiseFPException = 0 in 
+  defm FCVT_Q_L : FPUnaryOp_r_frm_m<0b1101011, 0b00010, Ext, Ext.PrimaryTy, 
```

Comment:
I think this should be `FPUnaryOp_r_frmlegacy_m`. No rounding is needed for this conversion.